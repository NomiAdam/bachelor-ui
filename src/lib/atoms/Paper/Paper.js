import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {lightTheme, darkTheme} from '../../constants/theme';

const StyledPaperWrapper = styled.div`
	padding: ${({padding}) => padding};
	border-color: transparent;
	border-radius: 3px;
	${props => props.width ? `min-width: ${props.width};` : ''}
	${props => props.height ? `min-height: ${props.height};` : ''} 
	box-shadow:  0 1px 5px ${darkTheme.darkGrey};
	background-color: ${({background}) => background}; 
	overflow: ${({ overflow }) => overflow};
	${({ center }) => center && `
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	`
    }
`;

const Paper = ({children, height, width, background, padding, center, overflow}) => (
    <StyledPaperWrapper
        height={height}
        width={width}
        background={background}
        padding={padding}
        center={center}
        overflow={overflow}
    >
        {children}
    </StyledPaperWrapper>
);

Paper.propTypes = {
    /**
     * Children nodes of Paper component
     */
    children: PropTypes.any,
    /**
     * String representation of Paper height, ie. 100%, 50px ....
     */
    height: PropTypes.string,
    /**
     * String representation of Paper width, ie. 100%, 50px ....
     */
    width: PropTypes.string,
    /**
     * String representation of background color
     */
    background: PropTypes.string,
    /**
     * String representation of css padding
     */
    padding: PropTypes.string,
    /**
     * Whether we should center according to flex box
     */
    center: PropTypes.bool,
    /**
     * String definition of css property overflow
     */
    overflow: PropTypes.string,
};

Paper.defaultProps = {
    background: lightTheme.white,
    overflow: 'hidden',
    padding: '10px',
};

export {StyledPaperWrapper}
export default Paper;
