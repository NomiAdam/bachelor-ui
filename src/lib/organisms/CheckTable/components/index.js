import TableBody from './TableBody';
import TableHeading from './TableHeading';

export {
    TableBody,
    TableHeading,
};
