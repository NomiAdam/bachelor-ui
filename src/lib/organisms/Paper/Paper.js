import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { darkTheme } from '../../constants/theme';
import DEFAULT_THEME from '../../utils/resolveTheme';

const StyledPaperWrapper = styled.div`
  ${ DEFAULT_THEME }
	padding: ${ ({ padding }) => padding };
	border-color: transparent;
	${ props => (props.width ? `min-width: ${ props.width };` : '') }
	${ props => (props.height ? `min-height: ${ props.height };` : '') } 
	box-shadow:  0 1px 5px ${ darkTheme.darkGrey };
	overflow: ${ ({ overflow }) => overflow };
	${ ({ center }) => center && `
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	`
}
`;

const Paper = ({
    children, height, width, backgroundColor, padding, center, overflow, borderRadius,
}) => (
    <StyledPaperWrapper
        height={ height }
        width={ width }
        backgroundColor={ backgroundColor }
        padding={ padding }
        center={ center }
        overflow={ overflow }
        borderRadius={ borderRadius }
    >
        {children}
    </StyledPaperWrapper>
);

Paper.propTypes = {
    /**
     * Children nodes of Paper component
     */
    children: PropTypes.any,
    /**
     * String representation of Paper height, ie. 100%, 50px ....
     */
    height: PropTypes.string,
    /**
     * String representation of Paper width, ie. 100%, 50px ....
     */
    width: PropTypes.string,
    /**
     * Optional String representation of background color
     */
    backgroundColor: PropTypes.string,
    /**
     * String representation of css padding
     */
    padding: PropTypes.string,
    /**
     * Whether we should center according to flex box
     */
    center: PropTypes.bool,
    /**
     * String definition of css property overflow
     */
    overflow: PropTypes.string,
    /**
   * Optional borderRadius
   */
    borderRadius: PropTypes.number,
};

Paper.defaultProps = {
    overflow: 'hidden',
    padding: '10px',
};

export { StyledPaperWrapper };
export default Paper;
