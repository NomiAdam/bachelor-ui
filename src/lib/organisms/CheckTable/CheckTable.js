import React from 'react';
import PropTypes from 'prop-types';
import Paper from '../../atoms/Paper/index';
import Table from '../../atoms/Table/index';
import {TableHeading, TableBody} from './components/index';
import {evolve, always, length} from 'ramda';

const CheckTable = ({selected, timeSheet, handleChange, isCheckbox}) => {

    const handleRowChange = (key) => (value) => {
        const field = selected[key];

        const evolvedField = evolve({value: always(value)}, field);
        const evolvedSelected = evolve({[key]: always(evolvedField)}, selected);

        handleChange(evolvedSelected);
    };

    const rowRange = length(timeSheet);

    return (
        <Paper overflow={'auto'}>
            <Table rowSpacing>
                <TableHeading dayHours={timeSheet}/>
                <TableBody
                    rowRange={rowRange}
                    handleChange={handleRowChange}
                    selected={selected}
                    isCheckbox={isCheckbox}
                />
            </Table>
        </Paper>
    );
};


CheckTable.propTypes = {
    /**
     * Array of possible day hours for current sport center, format below is expected
     * [ '8:00', '8:30', ... ]
     */
    timeSheet: PropTypes.array,
    /**
     * Object of current table Rows, format as below
     * { key: { label: 'label', value: [], disabled: [] }, ... }
     * value array is array of selected (checked) TableCells
     */
    selected: PropTypes.object,
    /**
     * Function triggered upon selected change, expect new object with changed values
     */
    handleChange: PropTypes.func,
    /**
     * Whether we are using this table using select boxes or not
     */
    isCheckbox: PropTypes.bool,
};

CheckTable.defaultProps = {
    timeSheet: [],
    selected: {},
    isCheckbox: true,
};

export default CheckTable;
