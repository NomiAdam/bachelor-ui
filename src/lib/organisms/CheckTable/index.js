import CheckTable from './CheckTable';

export default CheckTable;
