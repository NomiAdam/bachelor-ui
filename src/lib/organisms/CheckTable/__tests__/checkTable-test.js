/* eslint-disable no-undef */
import React from 'react';
import {shallow} from 'enzyme';
import CheckTable from '../CheckTable';
import Paper from '../../../atoms/Paper/index';

const mockChange = jest.fn();

const mockDayHour = [
    '06:00',
    '06:30',
    '07:00',
    '07:30',
    '08:00',
    '08:30',
];

const mockBody = {
    kurt1: {
        label: 'Kurt 1',
        value: [],
    },
    kurt2: {
        label: 'Kurt 2',
        value: [],
    }
};

describe('CheckTable', () => {
    it('should be defined', () => {
        expect(CheckTable).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<CheckTable />)).toMatchSnapshot();
    });
    it('should match the snapshot with data', () => {
        expect(shallow(<CheckTable timeSheet={mockDayHour} selected={mockBody} />)).toMatchSnapshot();
    });
    it('should be a of type Paper', () => {
        expect(shallow(<CheckTable />).type()).toBe(Paper);
    });
    it('should not trigger action during render', () => {
        shallow(<CheckTable handleChange={mockChange} />);
        expect(mockChange).not.toHaveBeenCalled();
    });
});
