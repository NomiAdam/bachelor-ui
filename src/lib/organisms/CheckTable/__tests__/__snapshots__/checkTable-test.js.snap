// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckTable should match the snapshot 1`] = `
<Paper
  background="#ffffff"
  overflow="auto"
  padding="10px"
  radius="3px"
>
  <Table
    rowSpacing={true}
  >
    <ReservationTableHead
      dayHours={Array []}
    />
    <ReservationTableBody
      dayHours={Array []}
      handleChange={[Function]}
      isCheckbox={true}
      rowRange={0}
      selected={Object {}}
    />
  </Table>
</Paper>
`;

exports[`CheckTable should match the snapshot with data 1`] = `
<Paper
  background="#ffffff"
  overflow="auto"
  padding="10px"
  radius="3px"
>
  <Table
    rowSpacing={true}
  >
    <ReservationTableHead
      dayHours={
        Array [
          "06:00",
          "06:30",
          "07:00",
          "07:30",
          "08:00",
          "08:30",
        ]
      }
    />
    <ReservationTableBody
      dayHours={Array []}
      handleChange={[Function]}
      isCheckbox={true}
      rowRange={6}
      selected={
        Object {
          "kurt1": Object {
            "label": "Kurt 1",
            "value": Array [],
          },
          "kurt2": Object {
            "label": "Kurt 2",
            "value": Array [],
          },
        }
      }
    />
  </Table>
</Paper>
`;
